### PR TITLE
Add events to cta clicks

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR adds event emitting when a CTA is clicked.

Depends on https://github.com/gravitational/teleport/pull/25073

Companion PR adding events to teleport.e CTAs: https://github.com/gravitational/teleport.e/pull/1242

### CODE REVIEW

File: api/proto/teleport/usageevents/v1/usageevents.proto [hunk 1]
```diff
@@ -194,6 +194,22 @@ message UIDiscoverDatabaseRDSEnrollEvent {
   int64 selected_resources_count = 4;
 }
 
+// UICallToActionClickEvent is emmited when a user clicks a Teleport Web UI's CTA.
+message UICallToActionClickEvent {
+  CTA cta = 2;
```
Comment (espadolini) [inline api/proto/teleport/usageevents/v1/usageevents.proto:199]: It doesn't matter, but why start with message number 2?

File: lib/usagereporter/teleport/types.go [hunk 2]
```diff
@@ -524,6 +538,12 @@ func ConvertUsageEvent(event *usageeventsv1.UsageEventOneOf, userMD UserMetadata
 		}
 
 		return ret, nil
+	case *usageeventsv1.UsageEventOneOf_UiCallToActionClickEvent:
+		return &UICallToActionClickEvent{
+			UserName: userMD.Username,
+			Cta:      prehogv1a.CTA(e.UiCallToActionClickEvent.Cta),
```
Comment (espadolini) [inline lib/usagereporter/teleport/types.go:544]: If this is what we're doing then we need a comment near the usageevents definition of CTA to remind us to keep it in sync with the one from prehogv1alpha.

File: web/packages/teleport/src/services/userEvent/types.ts [hunk 3]
```diff
@@ -110,6 +111,14 @@ export enum DiscoverEventStatus {
   Aborted = 'DISCOVER_STATUS_ABORTED', // user exits the wizard
 }
 
+export enum CtaEvents {
```
Comment (espadolini) [inline web/packages/teleport/src/services/userEvent/types.ts:114]: Isn't it possible to use the enum from `gen/proto/js/prehog/v1alpha/teleport_pb.js` here?

File: api/proto/teleport/usageevents/v1/usageevents.proto (hunk 1 above)
Comment (mcbattirola) [reply api/proto/teleport/usageevents/v1/usageevents.proto]: I think I copied a similar code from cloud and removed the field before without updating the cta number. Updated in https://github.com/gravitational/teleport/pull/25325/commits/3b7f9cdea7e5e01303f9a1be78ece95a9eb5bdae, thanks

File: web/packages/teleport/src/services/userEvent/types.ts (hunk 3 above)
Comment (mcbattirola) [reply web/packages/teleport/src/services/userEvent/types.ts]: It is, and its much better, thank you!
Added in https://github.com/gravitational/teleport/pull/25325/commits/b07bdf88feccdc3d9dda03e04a47a2df827f1967.

I've exported it from the types file, I think it makes sense. Let me know what you think.

File: lib/usagereporter/teleport/types.go (hunk 2 above)
Comment (mcbattirola) [reply lib/usagereporter/teleport/types.go:544]: Added some comments on `usageevents.proto`. Thanks.

**Review by espadolini — APPROVED** [review]

File: web/packages/teleport/src/services/userEvent/types.ts [hunk 4]
```diff
@@ -16,6 +16,11 @@
 
 // these constants are 1:1 with constants found in lib/usagereporter/web/userevent.go
 
+import * as prehogApi from 'gen-proto-js/prehog/v1alpha/teleport_pb';
+
+export type CtaEvents = prehogApi.CTA;
+export const CtaEvents = prehogApi.CTA;
```
Comment (gzdunek) [inline web/packages/teleport/src/services/userEvent/types.ts:22]: Exporting both type and value seems a bit weird (also I have never assigned `enum` to `const`😛), the more clean version would be:
```ts
export { CTA as CtaEvent } from 'gen-proto-js/prehog/v1alpha/teleport_pb';
```
Nit: it should be called `CtaEvent` (not `CtaEvents`), look at TS docs: https://www.typescriptlang.org/docs/handbook/enums.html#numeric-enums

Comment (mcbattirola) [reply web/packages/teleport/src/services/userEvent/types.ts:22]: Oh, this is totally weird. Sorry, I should've added a comment there.

So, even though `export { CTA as CtaEvent } from ...` passes the type check and tests, it fails on build:

```
"CTA" is not exported by "../../../gen/proto/js/prehog/v1alpha/teleport_pb.js", imported by "../../../web/packages/teleport/src/services/userEvent/types.ts".
```

It looks like we can only import * from `gen-proto-js/prehog/v1alpha/teleport_pb`.

If we do something like this:

```typescript
import * as prehogApi from 'gen-proto-js/prehog/v1alpha/teleport_pb';

export const CtaEvents = prehogApi.CTA;
```

`CtaEvents` is a value, not a type, so we can't use it as a type in functions:

```typescript
captureCtaEvent(event: CtaEvents) { // err: 'CtaEvents' refers to a value, but is being used as a type here. Did you mean 'typeof CtaEvents'?ts(2749)
}
```

And if we do `event: typeof CtaEvents`, we can't use the enum values:

```typescript
userEventService.captureCtaEvent(CtaEvents.CTA_ACCESS_REQUESTS); // err: Argument of type 'import("...v1alpha/teleport_pb").CTA' is not assignable to parameter of type 'typeof CTA'.ts(2345)
```

So AFAIK the only way to do that is with that strange hack I found on [stack overflow](https://stackoverflow.com/a/73624285/11411938):

> This way, we get the best of both worlds - types and values. And TS will figure out which one it wants.

Comment (mcbattirola) [reply web/packages/teleport/src/services/userEvent/types.ts]: Update: importing `gen-proto-js/prehog/v1alpha/teleport_pb` doesn't really work. It uses `require` and won't run in the browser.

I suppose there is a way to fix this (maybe generate the proto file with different flags)? But due to time constraints, I rather just copy the enum for now and go back to this later. Thoughts?

Comment (gzdunek) [reply web/packages/teleport/src/services/userEvent/types.ts]: Could you only leave a comment with TODO above the enum?

**Review by gzdunek — APPROVED** [review]

**Review by michellescripts — APPROVED** [review]

**Pull request merged**